src/dune_pkg/opamUrl0.ml

```diff
@@ -30,10 +30,16 @@ let is_version_control t =
 
 let is_local t = String.equal t.transport "file"
 
-let local_or_git_only url loc =
+let is_tarball t =
+  let supported_compress_format = [ "tar.gz"; "tgz"; "tar.bz2"; "tbz" ] in
+  List.exists ~f:(fun suffix -> String.is_suffix ~suffix t.path) supported_compress_format
```

Comment:
If you test with `String.is_suffix` then I think it would be better if the format list included `.` otherwise `"terminaltbz"` would be considered a tarball.

But the better solution is to use `Filename.check_suffix` since it does the right thing on e.g. Windows.